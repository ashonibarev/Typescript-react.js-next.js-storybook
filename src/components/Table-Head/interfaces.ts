type Order = 'asc' | 'desc'

export interface EnhancedTablePropsInterface {
  onRequestSort: (event: React.MouseEvent<unknown>, property: string) => void
  order: Order
  orderBy: string
  rowCount: number
  columns: Object[]
  loading: boolean
}
