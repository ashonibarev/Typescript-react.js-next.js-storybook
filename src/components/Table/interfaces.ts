import { ApiQuery } from 'lib/interfaces'
import { Filter, SetFilters } from '../Table-Toolbar/types'

export interface TableProps {
  title: string
  columns: any
  rows: Record<string, any>[]
  rowCount: number
  query: Partial<ApiQuery>
  setQuery: Function
  loading: boolean
  filters: Filter[]
  setFilters: SetFilters
  customComponents?: Record<string, Function>
  includeToolBar?: boolean
  expansionComponent?: Function
}
