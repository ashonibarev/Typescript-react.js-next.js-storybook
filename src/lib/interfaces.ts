import { NextApiRequest } from 'next'
import { Order } from './types'

export interface ApiQuery {
  searchTerm?: string
  page: number
  order: Order
  orderBy: string
  rowsPerPage: number
  filters?: string[][] | NextApiRequest['query']
}

// used for tables
export interface Column {
  id: string
  numeric: boolean
  disablePadding: boolean
  label: string
  type: string
}

export interface DrawRequest {
  id: string
  drawNumber: number
  loanId: number
  status: string
  requestedDate: Date
  approvedDate: Date
  fundedDate: string
  requestedAmount: string
  approvedAmount: string
  approvedAmountToDate: string
  disbursedDate: Date
  inspectionFee: string
  wireFee: string
  active: boolean
}

export interface DrawRequestGetResponse {
  drawRequest: DrawRequest
}

export interface InspectionVendor {
  id: number
  name: string
}

export interface Inspection {
  id: string
  status: string
  orderedDate: Date
  eta: Date
  receivedDate: Date
  inspectionDate: Date
  completePercentage: number
  approvedDate: Date
  vendorId: number
  active: boolean
  vendor: InspectionVendor
}

export interface Fci {
  loanId: number
  status: string
  assignment: string
  updatedAt: string
}

export interface FciGetResponse {
  fci: Fci
}

export interface LoanContract {
  lid: string
  number: string
  borrowerName: string
  amount: string
  fullAddress: string
  fciLoanId: number
}

export interface LoanContractGetResponse {
  loanContract: LoanContract
}

export interface LoanContractDrawInfo {
  approvedHoldback: string
  rehabBudget: string
  address: string
  city: string
  state: string
}

export interface LoanContractDrawInfoGetResponse {
  loanContractDrawInfo: LoanContractDrawInfo
}

export interface InspectionVendor {
  id: number
  name: string
}

export interface InspectionVendorListResponse {
  inspectionVendors: InspectionVendor[]
}
