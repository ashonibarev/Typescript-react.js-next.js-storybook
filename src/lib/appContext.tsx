import React, { createContext, useContext, useState } from 'react'
import { isEmpty } from 'lodash'

import useDrawerStats, { DrawerStats } from 'hooks/useDrawerStats'

type MessageLevel = 'success' | 'info' | 'warning' | 'error'

interface AppState {
  showMessage: boolean
  message: string
  messageLevel: MessageLevel
  setMessage: (message: string, messageLevel?: string) => void
  drawerStats: DrawerStats
}

export const AppContext = createContext<AppState>({} as AppState)

function AppContextProvider({ children }) {
  const drawerStats = useDrawerStats()

  const [state, setState] = useState({
    showMessage: false,
    message: '',
    messageLevel: 'success' as MessageLevel,
    setMessage,
  })

  function setMessage(message: string, messageLevel = 'success' as MessageLevel) {
    setState((prevState) => {
      return {
        ...prevState,
        message,
        showMessage: !isEmpty(message),
        messageLevel,
      }
    })
  }

  return <AppContext.Provider value={{ ...state, drawerStats }}>{children}</AppContext.Provider>
}

function useAppContext() {
  const context = useContext(AppContext)
  if (context === undefined) {
    throw new Error('useAccountingStore must be used within a AppContext Provider')
  }
  return context
}

export { useAppContext }

export default AppContextProvider
