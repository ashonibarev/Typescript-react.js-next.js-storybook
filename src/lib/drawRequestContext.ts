import React from 'react'

const DrawRequestContext = React.createContext(null)

export default DrawRequestContext
