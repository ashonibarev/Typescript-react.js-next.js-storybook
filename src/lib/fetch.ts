import 'isomorphic-unfetch'

interface PostResponse {
  [key: string]: any
}

export const post = async (
  url: string,
  data: BodyInit,
  options: Partial<RequestInit> = {},
): Promise<PostResponse> => {
  let res: Response
  // let payload: PostResponse = {}

  try {
    res = await fetch(url, {
      method: 'POST',
      mode: 'cors', // no-cors, *cors, same-origin
      cache: 'no-cache', // *default, no-cache, reload, force-cache, only-if-cached
      credentials: 'same-origin', // include, *same-origin, omit
      // headers: {
      //   'Content-Type': 'application/json',
      //   // 'Content-Type': 'multipart/form-data',
      // },
      redirect: 'follow', // manual, *follow, error
      referrerPolicy: 'no-referrer', // no-referrer, *no-referrer-when-downgrade, origin, origin-when-cross-origin, same-origin, strict-origin, strict-origin-when-cross-origin, unsafe-url
      body: data,
      ...options,
    })
  } catch (error) {
    throw new Error(error.message)
  }

  if (res.ok) {
    return res.json()
  }

  throw new Error(res.statusText)
}

export async function get(path, params) {
  var url = new URL(`/api/${path}`)
  url.search = new URLSearchParams(params).toString()
  return await fetch(url.toString())
}
