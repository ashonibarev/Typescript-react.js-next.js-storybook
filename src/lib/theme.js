import { createMuiTheme } from '@material-ui/core/styles'

// Create a theme instance.
const theme = createMuiTheme({
  palette: {
    type: 'light',
    primary: {
      // main: '#69f0ae',
      // main: '#42a5f5',
      main: '#455a64',
    },
    // secondary: {
    //   main: '#47dd92',
    // },
    // error: {
    //   main: red.A400,
    // },
  },
})

theme.shadows[1] = '0px 11px 15px 1px rgba(0, 0, 0, 0.02), 0px 4px 20px 3px rgba(0, 0, 0, 0.04)'
theme.shadows[2] =
  '00px 2px 1px -1px rgba(0,0,0,0.2), 0px 1px 1px 0px rgba(0,0,0,0.14), 0px 1px 3px 0px rgba(0,0,0,0.12)'

export default theme
