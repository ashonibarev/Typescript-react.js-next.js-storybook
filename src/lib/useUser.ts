import { useEffect } from 'react'
import Router from 'next/router'
import useSWR from 'swr'

interface Props {
  redirectTo: string
  redirectIfFound?: boolean
}

export default function useUser({ redirectTo, redirectIfFound = false }: Props) {
  const { data, mutate, error } = useSWR('/api/auth')

  const user = data?.user ?? null
  const finished = Boolean(data)
  const hasUser = Boolean(user)

  if (hasUser) {
    user.isExec = () => (user?.permissions ? user.permissions.includes('exec') : false)
    user.isAdmin = () => (user?.permissions ? user.permissions.includes('admin') : false)
  }

  useEffect(() => {
    if (!redirectTo || !finished) return
    if (
      // If redirectTo is set, redirect if the user was not found.
      (redirectTo && !redirectIfFound && !hasUser) ||
      // If redirectIfFound is also set, redirect if the user was found
      (redirectIfFound && hasUser)
    ) {
      Router.push(redirectTo)
    }
  }, [redirectTo, redirectIfFound, finished, hasUser])

  return { user, error, mutateUser: mutate }
}
