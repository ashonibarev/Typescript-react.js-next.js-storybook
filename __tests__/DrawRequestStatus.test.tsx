import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import {
  toBeInTheDocument,
  toHaveValue,
  toBeVisible,
  toHaveTextContent,
  toContainElement,
} from '@testing-library/jest-dom/matchers'

import DrawRequestStatus from 'components/DrawRequest/Status'
import { DrawRequest, Inspection } from 'lib/interfaces'
import useUser from 'lib/useUser'
import { act } from 'react-dom/test-utils'

const handleSubmit = (data: Partial<DrawRequest>) => {}
expect.extend({ toBeInTheDocument, toHaveValue, toBeVisible, toHaveTextContent, toContainElement })

jest.mock('lib/useUser', () => jest.fn())

beforeAll(() => {
  // @ts-ignore
  useUser.mockClear()
})

afterAll(() => {
  jest.resetAllMocks()
})

test('draw status `new`', async () => {
  const drawRequest: Partial<DrawRequest> = {
    status: 'new',
    requestedAmount: '$50,000',
    requestedDate: new Date('01/01/2020'),
  }

  const utils = render(
    <DrawRequestStatus drawRequest={drawRequest} allowedAmount={30000} onApprove={handleSubmit} />,
  )

  const stepNew = utils.getByTestId('stepper-new')

  // make sure step is there with correct label
  expect(stepNew).toBeInTheDocument()
  expect(stepNew).toHaveTextContent('new')
  expect(stepNew.children.length).toBe(2)

  expect(utils.container).toMatchSnapshot()

  // should have date, date should be correct
  const stepNewDate = utils.getByTestId('stepper-new-date')
  expect(stepNew).toContainElement(stepNewDate)
  expect(stepNewDate).toHaveTextContent('01.01.2020')

  // should not have the form
  expect(utils.container).not.toContainElement(utils.queryByLabelText('Approved Amount'))
  expect(utils.queryByTestId('approval-form-button')).not.toBeInTheDocument()

  // should have inspection ordered, but no date
  expect(utils.getByTestId('stepper-inspection ordered')).toBeInTheDocument()
  expect(utils.getByTestId('stepper-inspection ordered').children.length).toBe(1)
})

test('draw status `inspection ordered`', async () => {
  const drawRequest: Partial<DrawRequest> = {
    status: 'inspection ordered',
    requestedAmount: '$50,000',
    requestedDate: new Date('01/01/2020'),
  }

  const inspection: Partial<Inspection> = {
    orderedDate: new Date('01/02/2020'),
  }

  const { getByText, queryByTestId, queryByLabelText } = render(
    <DrawRequestStatus
      drawRequest={drawRequest}
      allowedAmount={30000}
      onApprove={handleSubmit}
      inspection={inspection}
    />,
  )

  expect(getByText('inspection ordered')).toBeInTheDocument()
  expect(getByText('01.02.2020')).toBeInTheDocument()
  expect(queryByTestId('approval-form-button')).not.toBeInTheDocument()
  expect(queryByLabelText('Approved Amount')).not.toBeInTheDocument()
})

test('draw status `inspection ordered`, basic without privilege', async () => {
  const drawRequest: Partial<DrawRequest> = {
    status: 'inspection approved',
    requestedAmount: '$50,000',
    requestedDate: new Date('01/01/2020'),
  }

  const inspection: Partial<Inspection> = {
    orderedDate: new Date('01/02/2020'),
    approvedDate: new Date('01/03/2020'),
  }

  const mockSubmit = jest.fn((data: Partial<DrawRequest>) => {})

  // @ts-ignore
  useUser.mockImplementation(() => ({
    user: {
      isExec: () => false,
    },
  }))

  const utils = render(
    <DrawRequestStatus
      drawRequest={drawRequest}
      allowedAmount={30000}
      onApprove={mockSubmit}
      inspection={inspection}
    />,
  )

  expect(utils.getByTestId('stepper-inspection approved')).toHaveTextContent('01.03.2020')

  expect(utils.getByText('inspection approved')).toBeInTheDocument()
  expect(utils.queryByTestId('approval-form-button')).toBeInTheDocument()
  expect(utils.queryByLabelText('Approved Amount')).toBeInTheDocument()

  // use input and test
  await act(async () => {
    fireEvent.change(utils.queryByLabelText('Approved Amount'), { target: { value: '30000' } })
  })

  expect(utils.queryByLabelText('Approved Amount')).toHaveValue('$30,000')

  await act(async () => {
    fireEvent.click(utils.queryByTestId('approval-form-button'))
  })

  expect(mockSubmit).toHaveBeenCalledWith({
    approvedAmount: '$30,000',
    approvedDate: expect.anything(),
    // This should be fine since we don't have overage
    status: 'approved',
  })
})

test('draw status `inspection ordered`, overage without privilege', async () => {
  const drawRequest: Partial<DrawRequest> = {
    status: 'inspection approved',
    requestedAmount: '$50,000',
    requestedDate: new Date('01/01/2020'),
  }

  const inspection: Partial<Inspection> = {
    orderedDate: new Date('01/02/2020'),
    approvedDate: new Date('01/03/2020'),
  }

  const mockSubmit = jest.fn((data: Partial<DrawRequest>) => {})

  // @ts-ignore
  useUser.mockImplementation(() => ({
    user: {
      isExec: () => false,
    },
  }))

  const utils = render(
    <DrawRequestStatus
      drawRequest={drawRequest}
      allowedAmount={30000}
      onApprove={mockSubmit}
      inspection={inspection}
    />,
  )

  expect(utils.getByTestId('stepper-inspection approved')).toHaveTextContent('01.03.2020')

  expect(utils.getByText('inspection approved')).toBeInTheDocument()
  expect(utils.queryByTestId('approval-form-button')).toBeInTheDocument()
  expect(utils.queryByLabelText('Approved Amount')).toBeInTheDocument()

  // use input and test
  await act(async () => {
    fireEvent.change(utils.queryByLabelText('Approved Amount'), { target: { value: '32000' } })
  })

  expect(utils.queryByLabelText('Approved Amount')).toHaveValue('$32,000')

  await act(async () => {
    fireEvent.click(utils.queryByTestId('approval-form-button'))
  })

  // Same as last test, but should NOT have date
  expect(mockSubmit).toHaveBeenCalledWith({
    approvedAmount: '$32,000',
  })
})

test('draw status `inspection ordered`, overage with privilege', async () => {
  const drawRequest: Partial<DrawRequest> = {
    status: 'inspection approved',
    requestedAmount: '$50,000',
    requestedDate: new Date('01/01/2020'),
  }

  const inspection: Partial<Inspection> = {
    orderedDate: new Date('01/02/2020'),
    approvedDate: new Date('01/03/2020'),
  }

  const mockSubmit = jest.fn((data: Partial<DrawRequest>) => {})

  // @ts-ignore
  useUser.mockImplementation(() => ({
    user: {
      isExec: () => true,
    },
  }))

  const utils = render(
    <DrawRequestStatus
      drawRequest={drawRequest}
      allowedAmount={30000}
      onApprove={mockSubmit}
      inspection={inspection}
    />,
  )

  expect(utils.getByTestId('stepper-inspection approved')).toHaveTextContent('01.03.2020')

  expect(utils.getByText('inspection approved')).toBeInTheDocument()
  expect(utils.queryByTestId('approval-form-button')).toBeInTheDocument()
  expect(utils.queryByLabelText('Approved Amount')).toBeInTheDocument()

  // use input and test
  await act(async () => {
    fireEvent.change(utils.queryByLabelText('Approved Amount'), { target: { value: '32000' } })
  })

  expect(utils.queryByLabelText('Approved Amount')).toHaveValue('$32,000')

  await act(async () => {
    fireEvent.click(utils.queryByTestId('approval-form-button'))
  })

  // Same as last test, but should NOT have date
  expect(mockSubmit).toHaveBeenCalledWith({
    approvedAmount: '$32,000',
    approvedDate: expect.anything(),
    // Should actually move to 'approved' status if person has permission
    status: 'approved',
  })
})
