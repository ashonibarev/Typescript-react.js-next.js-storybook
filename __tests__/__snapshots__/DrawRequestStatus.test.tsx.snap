// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`draw status \`new\` 1`] = `
<div>
  <div
    class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiCardHeader-root"
    >
      <div
        class="MuiCardHeader-content"
      >
        <span
          class="MuiTypography-root MuiCardHeader-title MuiTypography-subtitle2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
        >
          Status
        </span>
      </div>
      <div
        class="MuiCardHeader-action"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            hold
            <span
              class="MuiButton-endIcon MuiButton-iconSizeSmall"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M22 3H7c-.69 0-1.23.35-1.59.88L0 12l5.41 8.11c.36.53.9.89 1.59.89h15c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 16H7.07L2.4 12l4.66-7H22v14zm-11.59-2L14 13.41 17.59 17 19 15.59 15.41 12 19 8.41 17.59 7 14 10.59 10.41 7 9 8.41 12.59 12 9 15.59z"
                />
              </svg>
            </span>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      class="MuiCardContent-root"
    >
      <div
        class="makeStyles-root-108"
      >
        <div
          class="MuiPaper-root MuiStepper-root MuiStepper-horizontal MuiStepper-alternativeLabel MuiPaper-elevation0"
        >
          <div
            class="MuiStep-root MuiStep-horizontal MuiStep-alternativeLabel MuiStep-completed"
          >
            <span
              class="MuiStepLabel-root MuiStepLabel-horizontal MuiStepLabel-alternativeLabel"
            >
              <span
                class="MuiStepLabel-iconContainer MuiStepLabel-alternativeLabel"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-completed"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 0a12 12 0 1 0 0 24 12 12 0 0 0 0-24zm-2 17l-5-5 1.4-1.4 3.6 3.6 7.6-7.6L19 8l-9 9z"
                  />
                </svg>
              </span>
              <span
                class="MuiStepLabel-labelContainer"
              >
                <span
                  class="MuiTypography-root MuiStepLabel-label MuiStepLabel-alternativeLabel MuiStepLabel-completed MuiTypography-body2 MuiTypography-displayBlock"
                >
                  <div
                    class="Flex-e0vd0d-0 kLtFGV"
                    data-testid="stepper-new"
                    direction="column"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-caption"
                      data-testid="stepper-new-label"
                    >
                      new
                    </span>
                    <div
                      data-testid="stepper-new-date"
                    >
                      <div
                        class="makeStyles-root-137"
                      >
                        <span
                          class="MuiTypography-root MuiTypography-caption MuiTypography-colorTextSecondary"
                        >
                          01.01.2020
                        </span>
                      </div>
                    </div>
                  </div>
                </span>
              </span>
            </span>
          </div>
          <div
            class="MuiStep-root MuiStep-horizontal MuiStep-alternativeLabel"
          >
            <div
              class="MuiStepConnector-root MuiStepConnector-horizontal MuiStepConnector-alternativeLabel MuiStepConnector-active"
            >
              <span
                class="MuiStepConnector-line MuiStepConnector-lineHorizontal"
              />
            </div>
            <span
              class="MuiStepLabel-root MuiStepLabel-horizontal MuiStepLabel-alternativeLabel"
            >
              <span
                class="MuiStepLabel-iconContainer MuiStepLabel-alternativeLabel"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-active"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="12"
                  />
                  <text
                    class="MuiStepIcon-text"
                    text-anchor="middle"
                    x="12"
                    y="16"
                  >
                    2
                  </text>
                </svg>
              </span>
              <span
                class="MuiStepLabel-labelContainer"
              >
                <span
                  class="MuiTypography-root MuiStepLabel-label MuiStepLabel-alternativeLabel MuiStepLabel-active MuiTypography-body2 MuiTypography-displayBlock"
                >
                  <div
                    class="Flex-e0vd0d-0 kLtFGV"
                    data-testid="stepper-inspection ordered"
                    direction="column"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-caption"
                      data-testid="stepper-inspection ordered-label"
                    >
                      inspection ordered
                    </span>
                  </div>
                </span>
              </span>
            </span>
          </div>
          <div
            class="MuiStep-root MuiStep-horizontal MuiStep-alternativeLabel"
          >
            <div
              class="MuiStepConnector-root MuiStepConnector-horizontal MuiStepConnector-alternativeLabel Mui-disabled"
            >
              <span
                class="MuiStepConnector-line MuiStepConnector-lineHorizontal"
              />
            </div>
            <span
              class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled MuiStepLabel-alternativeLabel"
            >
              <span
                class="MuiStepLabel-iconContainer MuiStepLabel-alternativeLabel"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiStepIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="12"
                  />
                  <text
                    class="MuiStepIcon-text"
                    text-anchor="middle"
                    x="12"
                    y="16"
                  >
                    3
                  </text>
                </svg>
              </span>
              <span
                class="MuiStepLabel-labelContainer"
              >
                <span
                  class="MuiTypography-root MuiStepLabel-label MuiStepLabel-alternativeLabel MuiTypography-body2 MuiTypography-displayBlock"
                >
                  <div
                    class="Flex-e0vd0d-0 kLtFGV"
                    data-testid="stepper-inspection approved"
                    direction="column"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-caption"
                      data-testid="stepper-inspection approved-label"
                    >
                      inspection approved
                    </span>
                  </div>
                </span>
              </span>
            </span>
          </div>
          <div
            class="MuiStep-root MuiStep-horizontal MuiStep-alternativeLabel"
          >
            <div
              class="MuiStepConnector-root MuiStepConnector-horizontal MuiStepConnector-alternativeLabel Mui-disabled"
            >
              <span
                class="MuiStepConnector-line MuiStepConnector-lineHorizontal"
              />
            </div>
            <span
              class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled MuiStepLabel-alternativeLabel"
            >
              <span
                class="MuiStepLabel-iconContainer MuiStepLabel-alternativeLabel"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiStepIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="12"
                  />
                  <text
                    class="MuiStepIcon-text"
                    text-anchor="middle"
                    x="12"
                    y="16"
                  >
                    4
                  </text>
                </svg>
              </span>
              <span
                class="MuiStepLabel-labelContainer"
              >
                <span
                  class="MuiTypography-root MuiStepLabel-label MuiStepLabel-alternativeLabel MuiTypography-body2 MuiTypography-displayBlock"
                >
                  <div
                    class="Flex-e0vd0d-0 kLtFGV"
                    data-testid="stepper-approved"
                    direction="column"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-caption"
                      data-testid="stepper-approved-label"
                    >
                      approved
                    </span>
                  </div>
                </span>
              </span>
            </span>
          </div>
          <div
            class="MuiStep-root MuiStep-horizontal MuiStep-alternativeLabel"
          >
            <div
              class="MuiStepConnector-root MuiStepConnector-horizontal MuiStepConnector-alternativeLabel Mui-disabled"
            >
              <span
                class="MuiStepConnector-line MuiStepConnector-lineHorizontal"
              />
            </div>
            <span
              class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled MuiStepLabel-alternativeLabel"
            >
              <span
                class="MuiStepLabel-iconContainer MuiStepLabel-alternativeLabel"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiStepIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <circle
                    cx="12"
                    cy="12"
                    r="12"
                  />
                  <text
                    class="MuiStepIcon-text"
                    text-anchor="middle"
                    x="12"
                    y="16"
                  >
                    5
                  </text>
                </svg>
              </span>
              <span
                class="MuiStepLabel-labelContainer"
              >
                <span
                  class="MuiTypography-root MuiStepLabel-label MuiStepLabel-alternativeLabel MuiTypography-body2 MuiTypography-displayBlock"
                >
                  <div
                    class="Flex-e0vd0d-0 kLtFGV"
                    data-testid="stepper-funded"
                    direction="column"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-caption"
                      data-testid="stepper-funded-label"
                    >
                      funded
                    </span>
                  </div>
                </span>
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
